src/mq/gcloud-pubsub-push/mq.ts

```diff
@@ -312,30 +319,70 @@ export class GCloudPubSubPushMessageQueue implements MessageQueue {
                     pubSubId: message.id,
                     mq_message: message.data,
                     error,
+                    deliveryAttempt: deliveryAttemptCount,
                 },
             );
 
+            const errorAnalysis = analyzeError(error as Error);
+
             if (errorAnalysis.isReportable) {
                 this.errorListener?.(error as Error);
             }
 
-            if (shouldRetryUsingTopic && errorAnalysis.isRetryable) {
-                this.logger.info(
-                    `Publishing to retry topic [FedifyID: ${fedifyId}, PubSubID: ${message.id}]`,
+            if (
+                shouldRetryUsingTopic &&
+                errorAnalysis.isRetryable &&
+                deliveryAttemptCount < this.maxDeliveryAttempts
+            ) {
+                if (isFirstAttempt) {
```

Comment:
I think this condition should really be

> "If we're on the normal topic, publish to retry"
> "If we're on the retry topic, throw"

We're kinda inferring that from the trey count, which is maybe fine?

Reply:
🤔 yeh, it does feel a little off and might cause a wtf moment later if something changes, will update

Reply:
Have added an attribute to work out if the message is a retry or not

https://github.com/TryGhost/ActivityPub/pull/1090/commits/fc0b57b26d4dcfaa1fa4f9ff265c4fee47712823